File: 3058/Supermercado.py
t = int(input())

minor = 1000.00*1000

while t > 0:


    p, g = input().split()
    
    if (float(p)*1000)/int(g) < minor:
        minor = (float(p)*1000)/int(g)

    t-=1

print("%.2f"%(minor))

File: 3058/Supermercado.cpp
#include <bits/stdc++.h>
using namespace std;







int main (){

    int t; cin >> t;
    double minor = 1000.00*1000;

    double p;
    int g;

    while (t--){
        
        cin >> p >> g;
        

        if ((1000*p)/g < minor)
            minor = (1000*p)/g;


    }

    cout << setprecision(2) << fixed;
    cout << minor << endl;

}
